### PR TITLE
Flaky test fix: commentModerationTools.spec.js 

### LINKED ISSUE #19667: Flaky test: commentModerationTools.spec.js
## Is this feature related to a problem? Please describe.

Investigate `cypress/e2e/seededFlows/articleFlows/commentModerationTools.spec.js` it seems like the whole file is prone to flaky spec. We have unfortunately experienced an increase in flakiness and identified this test as one to take a look at more closely.

## Definition of Done

 - [ ] Ideally, a fix to resolve or at least reduce the flakiness for this test
 - [ ] If a fix cannot be determined, then a recommendation on what to do going forward

## Additional Context

- We want to timebox this investigation to no more than a week max.
- @maestromac noted that he is available to pull down the screenshot from failures and pair on debugging 
- https://buildpulse.io/@forem/forem

### PR DESCRIPTION
**Pull request by rt4914**
## What type of PR is this? (check all applicable)

- [x] Bug Fix

## Description

Flaky test fix.

Okay, so the tests were failing because there were few requests which were not finishing on time before executing the tests and because of which the test-based requests timed-out. 

To fix this @maestromac added a code to intercept and wait for all requests which should get executed before the test start. The reference used for this fix was: https://filiphric.com/how-to-wait-for-page-to-load-in-cypress

## Related Tickets & Documents

- Closes #19667

## QA Instructions, Screenshots, Recordings

Just run the `commentModerationTools ` cypress test locally multiple times to make sure that they do pass.

## Added/updated tests?

- [x] Yes

### CODE REVIEW

File: cypress/e2e/seededFlows/articleFlows/commentModerationTools.spec.js [hunk 1]
```diff
@@ -30,83 +23,84 @@ describe('Moderation Tools for Comments', () => {
 
       cy.get('@trustedUser').then((trustedUser) => {
         cy.loginAndVisit(trustedUser, '/series_user/series-test-article-slug');
+        // Reference: https://filiphric.com/how-to-wait-for-page-to-load-in-cypress
```
Comment (maestromac) [inline cypress/e2e/seededFlows/articleFlows/commentModerationTools.spec.js:26]: Typically, I would provide a link only to a solution to an edge case issue/fix. I don't think that's this case and would rather omit it.
```suggestion
```

File: cypress/e2e/seededFlows/articleFlows/commentModerationTools.spec.js [hunk 2]
```diff
@@ -30,83 +23,84 @@ describe('Moderation Tools for Comments', () => {
...
+        cy.intercept('/notifications/counts').as('notificationCount');
+        cy.intercept('/async_info/base_data').as('baseData');
+        cy.intercept('/ahoy/visits').as('ahoy');
+        cy.get('#comments').within(() => {
+          cy.findByRole('button', { name: 'Toggle dropdown menu' }).click();
+          cy.findByRole('link', { name: 'Moderate' }).click();
+        });
+        cy.wait(['@notificationCount', '@baseData', '@ahoy'], {
+          timeout: 10000,
+        });
```
Comment (maestromac) [inline cypress/e2e/seededFlows/articleFlows/commentModerationTools.spec.js:35]:
```suggestion
        cy.wait(['@notificationCount', '@baseData', '@ahoy']);
```
I rather use the default wait time than explicitly making it longer. If in the real world scenario, our users need to wait that long for the page to load, I rather we find out from test failure.

File: cypress/e2e/seededFlows/articleFlows/commentModerationTools.spec.js [hunk 3]
```diff
@@ -30,83 +23,81 @@ describe('Moderation Tools for Comments', () => {
 
       cy.get('@trustedUser').then((trustedUser) => {
         cy.loginAndVisit(trustedUser, '/series_user/series-test-article-slug');
+        cy.intercept('/notifications/counts').as('notificationCount');
+        cy.intercept('/async_info/base_data').as('baseData');
+        cy.intercept('/ahoy/visits').as('ahoy');
```
Comment (filleduchaos) [inline cypress/e2e/seededFlows/articleFlows/commentModerationTools.spec.js:28]: Ah - I'm a bit surprised that these weren't getting intercepted as they should? We're _supposed_ to be [automatically waiting for these to complete](https://github.com/forem/forem/blob/09828853f6cc0e07da74f4da885dfa93f5204662/cypress/util/networkUtils.js) whenever we call [`loginAndVisit`](https://github.com/forem/forem/blob/800ff397e0172fb0b165d76526c5803ab202cc22/cypress/support/commands.js#L45-L71)

**Review by filleduchaos — COMMENTED** [review]
Only commenting because I want to re-check the intercept behaviour in the morning before approving

**Review by filleduchaos — APPROVED** [review]
good short-term fix but there is a bigger problem where we currently don't have a way to automatically make Cypress wait for side effects like these requests when we navigate to a page through the app itself (e.g. clicking a link) vs when we call `cy.visit()` ourselves

But that's more for a follow-up issue

**Comment by rt4914** [comment]
> good short-term fix but there is a bigger problem where we currently don't have a way to automatically make Cypress wait for side effects like these requests when we navigate to a page through the app itself (e.g. clicking a link) vs when we call `cy.visit()` ourselves
> 
> But that's more for a follow-up issue

Thanks for this comment. It really sums-up this PR.

**Pull request merged**